json-functions: Update for MySQL 8.0 compat (#16859)
pingcap/docs: functions-and-operators/json-functions.md

# JSON Functions

> TiDB supports most of the JSON functions that shipped with the GA release of MySQL 8.0.

Reviewer: ```suggestion
TiDB supports most of the [JSON functions](https://dev.mysql.com/doc/refman/8.0/en/json-functions.html) available in MySQL 8.0.
```

| Function Name                     | Description |
| --------------------------------- | ----------- |
| [JSON_APPEND(json_doc, path, value)](https://dev.mysql.com/doc/refman/8.0/en/json-modification-functions.html#function_json-append) | An alias to `JSON_ARRAY_APPEND` |

> | [JSON_ARRAY_APPEND(json_doc, path, val[, path, val] ...)](https://dev.mysql.com/doc/refman/8.0/en/json-modification-functions.html#function_json-array-append) | Appends values to the end of the indicated arrays within a JSON document and returns the result |

Reviewer: Tip for reviewers: L38 and L39 were L48 and L49 in the original doc

> | [JSON_ARRAY_INSERT(json_doc, path, val[, path, val] ...)](https://dev.mysql.com/doc/refman/8.0/en/json-modification-functions.html#function_json-array-insert) | Insert values into the specified location of a JSON document and returns the result |

Reviewer 2: Maybe be consistent with singular or plural of `arrays` vs `location`, since both seems to take one or more paths?
I.e. either have both `indicated array` and `specified location` *or* `indicated arrays` and `specified locations`?

Author: @qiancai I also noticed `JSON document` vs `json document` here. Which one should it be?

Reviewer: `JSON document` is more proper as `JSON` is the correct format name  (see https://www.json.org/json-en.html)

| [JSON_INSERT(json_doc, path, val[, path, val] ...)](https://dev.mysql.com/doc/refman/8.0/en/json-modification-functions.html#function_json-insert) | Inserts data into a JSON document and returns the result |
| [JSON_MERGE_PATCH(json_doc, json_doc[, json_doc] ...)](https://dev.mysql.com/doc/refman/8.0/en/json-modification-functions.html#function_json-merge-patch)  | Merge JSON documents |
| [JSON_MERGE_PRESERVE(json_doc, json_doc[, json_doc] ...)](https://dev.mysql.com/doc/refman/8.0/en/json-modification-functions.html#function_json-merge-preserve)  | Merges two or more JSON documents and returns the merged result |

> | [JSON_MERGE(json_doc, json_doc[, json_doc] ...)](https://dev.mysql.com/doc/refman/8.0/en/json-modification-functions.html#function_json-merge)  | A deprecated alias for `JSON_MERGE_PRESERVE` |

Reviewer: Tip for reviewers: L43  is L41 in the original doc

* [JSON Function Reference](https://dev.mysql.com/doc/refman/8.0/en/json-function-reference.html)
* [JSON Data Type](/data-type-json.md)

## Unsupported functions

- `JSON_SCHEMA_VALID()`
- `JSON_SCHEMA_VALIDATION_REPORT()`
- `JSON_TABLE()`
- `JSON_VALUE()`

> See also [GitHub Issue #14486](https://github.com/pingcap/tidb/issues/14486)

Reviewer: ```suggestion
For more information, see [#14486](https://github.com/pingcap/tidb/issues/14486).
```
